### PR TITLE
Mask API key for Aleph Alpha LLM

### PR DESCRIPTION
**Pull request by slangenbach**
- **Description:** Add masking of API Key for Aleph Alpha LLM when printed.
- **Issue**: #12165
- **Dependencies:** None
- **Tag maintainer:** @eyurtsev

### CODE REVIEW

File: libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"""Test Aleph Alpha specific stuff."""
+
+from pytest import CaptureFixture, MonkeyPatch
```
Comment (eyurtsev) [inline libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py:3]: neat! didn't know about `CaptureFixture`!

File: libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+from langchain.llms.aleph_alpha import AlephAlpha
+from langchain.pydantic_v1 import SecretStr
+
+
```
Comment (eyurtsev) [inline libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py:8]: Could you skip these tests if aleph-alpha is not installed? Most langchain dependencies are optional. 

Two options: 

1) Add the dependency to extended testing extra in pyproject.toml , and then use the langchain custom @pytest.mark.requires marker. This marker will skip it in regular unit tests, but run the tests on CI in extended unit tests where this dependency will be installed
2) Use pytest skip to skip running the entire module if the alpeh alpha package is missing

**Review by eyurtsev — COMMENTED** [review]
Looks great! we just need to conditionally skip the test

File: libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py (hunk 2 above)
Comment (slangenbach) [reply libs/langchain/tests/unit_tests/llms/test_aleph_alpha.py:8]: Sure. I'll implement the first option later today.

**Review by eyurtsev — APPROVED** [review]
Looks great!

**Comment by eyurtsev** [comment]
merge when tests pass

**Comment by eyurtsev** [comment]
@slangenbach i'll resolve conflicts and relock poetry file

**Pull request merged**